### DISCUSSION TITLE
A little help with good search functionality

### DISCUSSION THREAD

**Discussion by DennisNissenUX** [Budibase/budibase#7356] [Help]
Hey guys,

I have some questions about the best practices for the search functionalities in tables.

First of, in this case I am using budibase internal datasources.
In one case I have a row with multiple relationships that I want to search inside. I know that there is no real support for filtering of relationship values in the tables, but this functionality would be very important for my use case here so I'm looking for a good workaround. Does anyone have experience with this or has an idea on how to achieve this? I was thinking about making a static formula with just a string that lists the names of the relationships and then searching through it with the chosen string/value from my dropdown picker. My problem is, that I didn't find a way to implement this kind of search/filter behaviour.

Another (maybe related) topic is the search of text, like the name column. It only seems to work with the beginning of the name, or the even just the exact same search statement like the value of the row. Is there a way to search for "include" like in the example below (search for "Nissen" in "Dennis Nissen der Dritte"),

Thank you very much in advance and have a great weekend!

![Bildschirmfoto 2022-08-19 um 09 34 02](https://user-images.githubusercontent.com/105709029/185568144-7fd2242d-bbee-4c73-9af4-492668cf8a0b.png)

![Bildschirmfoto 2022-08-19 um 09 38 24](https://user-images.githubusercontent.com/105709029/185570081-ce955987-8d3b-489c-ab56-9d135b7a665d.png)

**Comment by andz-bb** [comment]
Hey Dennis, unfortunately I wasn't able to get the relationship filtering working with the many-many setup you are using there, although I was able to set up a filter on a one-many relationship using the workaround described in the answer to #5661. I don't think it's possible yet to use that for cases where there are multiple values in the relationship column, but I have a feeling #6815 could be used to solve that. It has been merged so hopefully should be released soon!

For your second question I don't see an easy way to do it right now either, but it's possible the PR mentioned above could help with that functionality too. What I'm thinking is you could split the name string into an array of strings for each part of the name and then see if that array contains the search term. I think that update should be coming soon so hopefully it will be useful in both of these cases!
